src/workloads/query/plan_cache/MatchEqVaryingArray.yml

```diff
@@ -0,0 +1,120 @@
+Owner: "@mongodb/query"
+Description: |
+  This test was created to demonstrate a problem with the hit rate of the SBE plan cache.
+
+  It runs a query like {$match: {a: {$eq: [1]}}} where the number varies. The Classic plan
+  cache is able to reuse the same plan even as the parameter varies, but the SBE plan cache
+  treats each one separately, resulting in much more planning.
+
+  TODO more nonselective indexed predicates to increase planning time?
+  TODO more docs to increase planning time?
+
+
+GlobalDefaults:
+  dbname: &db test
+  # This name is not actually something we choose: the Loader actor chooses it.
+  coll: &coll Collection0
+
+  docCount: &docCount 1000
+  maxPhase: &maxPhase 7
+  queryRepeats: &queryRepeats 1000
+
+Actors:
+  - Name: Drop
+    Type: RunCommand
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [0]
+        NopInPhasesUpTo: *maxPhase
+        PhaseConfig:
+          Repeat: 1
+          Database: *db
+          Operations:
+            - OperationName: RunCommand
+              OperationCommand:
+                drop: *coll
+            # Explicitly create the collection, to avoid a special case where empty collections
+            # give you an EOF plan.
+            - OperationName: RunCommand
+              OperationCommand:
+                create: *coll
+
+  - Name: CreateDataset
+    Type: Loader
+    Threads: 1
+    Database: *db  # TODO unused?
+    Phases:
+      OnlyActiveInPhases:
+        Active: [1]
+        NopInPhasesUpTo: *maxPhase
+        PhaseConfig:
+          Threads: 1
+          Repeat: 1
+          Database: *db  # TODO unused?
+          Collection: *coll  # TODO unused?
+          CollectionCount: 1
+          DocumentCount: *docCount
+          BatchSize: 1000
+          Document: {
+            x00: {^Inc: {}},
+            x01: {^Inc: {}},
+          }
+          Indexes:
+            - keys: {x00: 1}
+            - keys: {x01: 1}
+
+  - Name: Quiesce
+    Type: QuiesceActor
+    Threads: 1
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: [2]
+        NopInPhasesUpTo: *maxPhase
+        PhaseConfig:
+          Repeat: 1
+
+  - Name: MatchQuery02Indexes
+    Type: CrudActor
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [3]
+        NopInPhasesUpTo: *maxPhase
+        PhaseConfig:
+          Repeat: *queryRepeats
+          Database: *db
+          Collection: *coll
+          Operations:
+            - OperationName: aggregate
+              OperationCommand:
+                Pipeline: [
+                  {$match: {
+                    # The main predicate which we vary to show the problem with auto-parameterization.
+                    a: {$eq: [ {^Inc: {}}, ]},
+
+                    # Other unindexed predicates which we include only to increase the cost of planning.
+                    # Multiplanning only happens if there are at least two indexed predicates.
+                    x00: {$gte: 0},
```

Comment:
I'd say add a few more of these so we do really capture the "worst case" (I think you have a TODO about this anyway though)

Reply:
I'll add a few variations: 2, 4, 8 indexed-but-nonselective predicates.